src/sp/transport/mqtt/broker_tcp.c

```diff
@@ -1067,12 +1067,14 @@ nmq_pipe_send_start_v4(tcptran_pipe *p, nni_msg *msg, nni_aio *aio)
 			// APP layer must give topic name even if topic
 			// alias is set
 			if (qos_pac > 0) {
-				property_len = get_var_integer(
-				    body + 4 + tlen, &property_bytes);
+				mqtt_get_remaining_length(body + 4 + tlen - 1,
+				    (uint32_t) nni_msg_header_len(msg),
+				    &property_len, &property_bytes);
 
 			} else {
-				property_len = get_var_integer(
-				    body + 2 + tlen, &property_bytes);
+				mqtt_get_remaining_length(body + 2 + tlen - 1,
```

Comment:
NO!!! The function name is not quite proper.

Reply:
The function is not conformed to its usage

Reply:
`get_remaining_length` can better illustrate its usage than `get_var_integer`, isn't it?

Reply:
Especially, it has a prefix `mqtt_`